Oras/Remote/Repository.cs

```diff
@@ -0,0 +1,991 @@
+
+
+                }
+                if (last != "")
+                {
+                    query["last"] = last;
+                }
+            }
+
+            uriBuilder.Query = query.ToString();
+            using var resp = await HttpClient.GetAsync(uriBuilder.ToString(), cancellationToken);
+            if (resp.StatusCode != HttpStatusCode.OK)
+            {
+                throw await ErrorUtility.ParseErrorResponse(resp);
+
+            }
+            var data = await resp.Content.ReadAsStringAsync();
+            var tagList = JsonSerializer.Deserialize<ResponseTypes.TagList>(data);
+            fn(tagList.Tags);
+            return LinkUtility.ParseLink(resp);
+        }
+
+        /// <summary>
+        /// DeleteAsync removes the content identified by the descriptor in the
+        /// entity blobs or manifests.
+        /// </summary>
+        /// <param name="target"></param>
+        /// <param name="isManifest"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        /// <exception cref="NotFoundException"></exception>
+        /// <exception cref="Exception"></exception>
+        internal async Task DeleteAsync(Descriptor target, bool isManifest, CancellationToken cancellationToken)
+        {
+            var remoteReference = RemoteReference;
+            remoteReference.Reference = target.Digest;
+            string url;
+            if (isManifest)
+            {
+                url = URLUtiliity.BuildRepositoryManifestURL(PlainHTTP, remoteReference);
+            }
+            else
+            {
+                url = URLUtiliity.BuildRepositoryBlobURL(PlainHTTP, remoteReference);
+            }
+
+            using var resp = await HttpClient.DeleteAsync(url, cancellationToken);
+
+            switch (resp.StatusCode)
+            {
+                case HttpStatusCode.Accepted:
+                    VerifyContentDigest(resp, target.Digest);
+                    break;
+                case HttpStatusCode.NotFound:
+                    throw new NotFoundException($"digest {target.Digest} not found");
+                default:
+                    throw await ErrorUtility.ParseErrorResponse(resp);
+            }
+        }
+
+
+        /// <summary>
+        /// VerifyContentDigest verifies "Docker-Content-Digest" header if present.
+        /// OCI distribution-spec states the Docker-Content-Digest header is optional.
+        /// Reference: https://github.com/opencontainers/distribution-spec/blob/v1.0.1/spec.md#legacy-docker-support-http-headers
+        /// </summary>
+        /// <param name="resp"></param>
+        /// <param name="expected"></param>
+        /// <exception cref="NotImplementedException"></exception>
+        internal static void VerifyContentDigest(HttpResponseMessage resp, string expected)
+        {
+            if (resp.Content.Headers.TryGetValues("Docker-Content-Digest", out var digestValues) is var gotValues && !gotValues) return;
```

Comment:
Is this equivalent to

```suggestion
            if (!resp.Content.Headers.TryGetValues("Docker-Content-Digest", out var digestValues))
```
?

Reply:
yh